distributed/comm/__init__.py

```diff
@@ -14,6 +14,7 @@
 
 
 def _register_transports():
+    from . import mpi
```

Comment:
This should probably be in a `try...except...` like `ucx` below as `mpi4py` is likely not always available. Also would suggest moving this next to `ucx` below since `inproc` and `tcp` are always available and should probably be registered first

Reply:
Makes sense.